source/adminguide/templates/_cloud_init.rst

```diff
@@ -75,15 +65,15 @@ These features can be implemented in `“Linux template creation process” <_cr
 	   
 #. **Password management**
 
-   Cloudstack integration with cloud-init `set-passwords module <https://cloudinit.readthedocs.io/en/latest/topics/modules.html?highlight=ssh_pwauth#set-passwords>`_ will enable the platform to set a password for each instance created from the Main Template. Additionally it will allow to reset the user password through the GUI.
+   Cloudstack integration with cloud-init `set_passwords module <https://cloudinit.readthedocs.io/en/latest/topics/modules.html?highlight=ssh_pwauth#set-passwords>`_ will enable the platform to set a password for each instance created from the Main Template. Additionally it will allow to reset the user password through the GUI.
```

Comment:
@TheRealFalcon 
are you saying that cloud-init can get the new password from cloudstack VR (password server) ?
if it works, it would be awesome.

Reply:
I become aware it from this documentation!

But based on the code (I haven't tested it personally), it appears that's what happens.

Reply:
good to know it. thanks